lib/slack-ruby-bot/mvc/controller/base.rb

```diff
@@ -0,0 +1,111 @@
+module SlackRubyBot
+  module MVC
+    module Controller
+      class Base
+        include ActiveSupport::Callbacks
+
+        class << self
+          attr_reader :abstract
+          alias abstract? abstract
+
+          def controllers
+            Base.instance_variable_get(:@controllers)
```

Comment:
Maybe you can declare an accessor for the base version, then use normal methods. I am not sure that's better though, so feel free to ignore me.